lib/event-handler.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const {
+  getProperty
+, setProperty
+} = require('./utils.js')
+
+module.exports = {
+  handleEvent
+}
+
+function handleEvent(event) {
+  let record = getProperty(event, 'Records.0')
+  if (!record) return undefined
+
+  let key = getProperty(record, 's3.object.key')
+  if (key) {
+    key = key.replace(/\+/g, ' ')
+    record = setProperty(record, 's3.object.key', decodeURIComponent(key))
+  }
+
+  let timestamp = Date.now()
+  if (record.eventTime) {
+    timestamp = (new Date(record.eventTime)).getTime()
+  }
```

Comment:
not sure if the date format is well-defined here, but if not it would be safer to avoid `NaN` values:

```js

if (record.eventTime) {
  const eventTimestamp = (new Date(record.eventTime)).getTime()
  if (!isNaN(eventTimestamp)) timestamp = eventTimestamp
}

Reply:
Well, it will always parse per [docs](https://docs.aws.amazon.com/lambda/latest/dg/with-s3.html) but let me add for the sake of safety